libsolidity/codegen/CompilerUtils.h

```diff
@@ -169,8 +169,12 @@ class CompilerUtils
 	/// Stack post:
 	void storeStringData(bytesConstRef _data);
 
-	/// Appends code that cleans higher-order bits for integer types.
-	void cleanHigherOrderBits(IntegerType const& _typeOnStack);
+	/// Helper function for shifting values of fixed points to correct value while changing types eg fixed56x56 -> fixed128x128
+	/// essentially just moving an imaginary decimal point.
+	void shiftFixedPointNumber(FixedPointType const& _stackType, FixedPointType const& _targetType);
```

Comment:
`appendShiftOperator` is more expensive because it deals with two items on the stack, while yours only has one stack item (before pushing your shift factor).